### PR TITLE
[WIP]: support modern bpf probe into the helm chart as a new driver

### PR DESCRIPTION
**Pull request by Andreagit97**
**What type of PR is this?**

/kind feature

/kind chart-release

**Any specific area of the project related to this PR?**

/area falco-chart

**What this PR does / why we need it**:

This PR adds support to the modern bpf probe as a new driver in the helm chart. Main changes:
* new driver.kind -> " modern_bpf"
* no init container for the modern probe
* only full privileged mode (no least privileged mode at least for now)
* new Falco config in the Falco yaml section `modern_bpf.cpus_for_each_syscall_buffer`

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

I've already bumped the Falco version to 0.34 but not sure how we want to proceed. And also not sure about the new chart version. We have to discuss a little bit about it :) 

**Checklist**

- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] CHANGELOG.md updated

### CODE REVIEW

**Comment by Andreagit97** [comment]
I've already bumped the Falco version to 0.34 but not sure how we want to proceed. And also not sure about the new chart version. We have to discuss a little bit about it :)

File: falco/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
 name: falco
-version: 2.5.2
-appVersion: 0.33.1
+version: 2.6.0
+appVersion: 0.34.0
```
Comment (Andreagit97) [inline falco/Chart.yaml:4]: we can discuss about these versions

File: falco/README.md [hunk 2]
```diff
@@ -47,12 +47,16 @@ Starting from Falco 0.31.0 the [new plugin system](https://falco.org/docs/plugin
...
 
-Falco needs a **driver** (the [kernel module](https://falco.org/docs/event-sources/drivers/#kernel-module) or the [eBPF probe](https://falco.org/docs/event-sources/drivers/#ebpf-probe)) that taps into the stream of system calls and passes that system calls to Falco. The driver must be installed on the node where Falco is running.
+#### About Drivers
 
-By default the drivers are managed using an *init container* which includes a script (`falco-driver-loader`) that either tries to build the driver on-the-fly or downloads a prebuilt driver as a fallback. Usually, no action is required.
+Falco can use different drivers to analyze the system workload and pass security events to userspace. The supported drivers are:
 
+* [Kernel module](https://falco.org/docs/event-sources/drivers/#kernel-module) 
+* [eBPF probe](https://falco.org/docs/event-sources/drivers/#ebpf-probe)
+* [Modern eBPF probe]() (starting from Falco `0.34.0`)
```
Comment (Andreagit97) [inline falco/README.md:57]: still miss the link to the docs, right now we don't have any doc regarding this :/

File: falco/templates/pod-template.tpl [hunk 3]
```diff
@@ -371,7 +374,7 @@ spec:
 {{- define "falco.securityContext" -}}
 {{- $securityContext := dict -}}
 {{- if .Values.driver.enabled -}}
-  {{- if eq .Values.driver.kind "module" -}}
+  {{- if or (eq .Values.driver.kind "module") (eq .Values.driver.kind "modern-bpf") -}}
```
Comment (Andreagit97) [inline falco/templates/pod-template.tpl:377]: i've just enabled the privileged mode as first step, we can do better in next steps

**Review by leogr — COMMENTED** [review]
Since this and `falcoctl` and other small changes will come all together along with Falco 0.34, I really believe it's more convenient to merge this PR into https://github.com/falcosecurity/charts/pull/453.

So we will have just a single PR that includes all changes related to Falco 0.34, and we will release all those changes together.

File: falco/values.yaml [hunk 4]
```diff
@@ -58,7 +58,7 @@ podSecurityContext: {}
 # 1) driver.enabled = false:
 #    securityContext: {}
 #
-# 2) driver.enabled = true and driver.kind = module:
+# 2) driver.enabled = true and (driver.kind = module || driver.kind = modern-bpf):
```
Comment (leogr) [inline falco/values.yaml:61]:
```suggestion
# 2) driver.enabled = true and (driver.kind = module || driver.kind = modern-bpf):
```

this should be a separate point

**Comment by leogr** [comment]
Hey @Andreagit97 

As discussed privately, I've cherry-picked relevant commits into https://github.com/falcosecurity/charts/pull/453
So, I'd close this PR if that works for you.

Thank you!

**Comment by Andreagit97** [comment]
thank you!

**Pull request closed without merging**